appinventor/components/src/com/google/appinventor/components/runtime/ActivityStarter.java

```diff
@@ -425,6 +453,15 @@ private Intent buildActivityIntent() {
       intent.putExtra(extraKey, extraValue);
     }
 
+    for (Object extra : extras.toArray()) {
+      YailList castExtra = (YailList) extra;
+      String key = castExtra.getString(0);
```

Comment:
yes, it works. I tried to log the key and value variable by adding 
`Log.d("ActivityStarter", "key: " + key + ", value: " + value);`
in the for-loop and they were printed correctly